javascript/ql/src/Security/CWE-327/BadRandomness.qhelp

```diff
@@ -4,33 +4,57 @@
 <qhelp>
 	<overview>
 		<p>
-			Placeholder
+			Generating secure random numbers can be an important part of creating
+			a secure software system, and for that purpose there exists secure APIs
+			for creating cryptographically secure random numbers.  
+		</p>
+		<p>
+			However, using some mathematical operations on these cryptographically 
+			secure random numbers can create biased results, where some outcomes 
+			are more likely than others. 
+			Such biased results can make it easier for an attacker to guess the random
+			numbers, and thereby break the security of the software system. 
 		</p>
-
 	</overview>
 	<recommendation>
-
 		<p>
-			Placeholder.
+			Be very careful not to introduce bias when performing mathematical operations
+			on cryptographically secure random numbers.  
+		</p>
+		<p>
+			If possible, avoid performing mathematical operations on cryptographically secure 
+			random numbers at all, and use a preexisting library instead. 
 		</p>
-
 	</recommendation>
 	<example>
-
 		<p>
-			Placeholder
+			The below example uses the modulo operator to create an array of 10 random digits
+			using random bytes as the source for randomness. 
 		</p>
+		<sample src="examples/bad-random.js" />
+		<p>
+			The random byte is a uniformly random value between 0 and 255, and thus the result 
+			from using the modulo operator is slightly more likely to be between 0 and 5 than 
+			between 6 and 9.
+		</p>
+		<p>
+			The issue has been fixed in the code below, where the random byte is discarded if 
+			the value was greater than or equal to 250. 
+			Thus the modulo operator is used on a uniformly random number between 0 and 249, which
```

Comment:
This sentence is a bit difficult to parse (I think it's because there's a lot of repetition of "uniformly random") but ignore my comment if you think developers will understand.

Reply:
In that sentence I wanted to emphasize that both the input and output are "uniformly random", and I haven't found a better way than repeating "uniformly random". 
I think developers will understand. 

I could change the example to use a library, instead of fixing the bug in the math.   
Actually, I think I'll add such an example as a first fix, and keep this fix as the second fix.